.github/workflows/compile-macos-arm64.yaml

```diff
@@ -0,0 +1,35 @@
+name: macOS(ARM64)
+
+on: [pull_request]
+
+jobs:
+  compile-macOS-ARM64:
+    runs-on: macos-latest
+    steps:
+    - name: Check Platform
+      run: |
+        if [ ! $(uname -m) = "arm64" ]; then
+            echo "Not on ARM64" >&2
+            exit 1
+        fi
```

Comment:
This check really isn't necessary.

Reply:
Github is the owner of this runner.

Reply:
BTW, you specifically chose a variable name for the runner -- `macos-latest`.  From the docs, I'm guessing that name moves to whatever the latest macOS runner is.  If you pick a specific runner, it will likely stay that architecture for the life of that runner.

That being said, per what I said above: if you want to have a check that is purely defensive in nature, that's fine, but you need to put a comment explaining *why*.

Reply:
> they changed from intel to arm

GitHub did that simply because Apple is phasing out Intel hardware and x86_64. I don't see Apple going the other way any time soon. Therefore, I think it is safe to assume that the `macos-latest` runner will stay in arm64 for many years to come.